jpsonic-main/src/main/java/com/tesshu/jpsonic/service/upnp/processor/SongByGenreProc.java

```diff
@@ -34,17 +39,27 @@
 @Service
 public class SongByGenreProc extends DirectChildrenContentProc<Genre, MediaFile> {
 
+    private static final MediaType[] TYPES = { MediaType.MUSIC, MediaType.AUDIOBOOK };
+
+    private final SettingsService settingsService;
     private final SearchService searchService;
     private final UpnpDIDLFactory factory;
     private final UpnpProcessorUtil util;
 
-    public SongByGenreProc(UpnpProcessorUtil util, UpnpDIDLFactory factory, SearchService searchService) {
+    public SongByGenreProc(SettingsService settingsService, UpnpProcessorUtil util, UpnpDIDLFactory factory,
+            SearchService searchService) {
         super();
+        this.settingsService = settingsService;
         this.util = util;
         this.factory = factory;
         this.searchService = searchService;
     }
 
+    private GenreMasterCriteria createGenreMasterCriteria() {
+        return new GenreMasterCriteria(util.getGuestFolders(), Scope.SONG,
+                Sort.of(settingsService.getUPnPSongGenreSort()), TYPES);
+    }
+
     @Override
```

Comment:
SongByGenreProc : The rewrite is insufficient. Rewrite it so that the previous GenreMaster is no longer used.

Reply:
Done.